File: tests/adapters/rust/goldens/function_bodies/basic_strip.rs
// Rust module for testing function body optimization.

use std::collections::HashMap;

#[derive(Debug, Clone)]
struct User {
    id: i32,
    name: String,
    email: String,
}

#[derive(Debug)]
struct ProcessingResult {
    valid: Vec<User>,
    invalid: Vec<User>,
}

struct Calculator {
    name: String,
    history: Vec<String>,
}

impl Calculator {
    fn new() -> Self // … method body omitted (3 lines)

    fn with_name(name: &str) -> Self // … method body omitted (6 lines)

    fn add(&mut self, a: i32, b: i32) -> i32 // … method body omitted (7 lines)

    fn multiply(&mut self, a: i32, b: i32) -> i32 // … method body omitted (6 lines)

    fn get_history(&self) -> Vec<String> // … method body omitted (3 lines)

    fn validate_input(&self, value: i32) -> bool // … method body omitted (17 lines)
}

fn process_user_data(users: Vec<User>) -> ProcessingResult // … function body omitted (16 lines)

fn process_array<T, F>(items: Vec<T>, processor: F) -> Vec<T>
where
    F: Fn(T) -> T,
// … function body omitted (10 lines)

fn filter_positive(numbers: Vec<i32>) -> Vec<i32> // … function body omitted (3 lines)

fn main() // … function body omitted (21 lines)

File: tests/adapters/cpp/goldens/function_bodies/basic_strip.cpp
/**
 * C++ module for testing function body optimization.
 */

#include <iostream>
#include <vector>
#include <string>
#include <algorithm>

struct User {
    int id;
    std::string name;
    std::string email;
};

struct ProcessingResult {
    std::vector<User> valid;
    std::vector<User> invalid;
};

class Calculator {
private:
    std::vector<std::string> history;
    std::string name;

public:
    Calculator() : Calculator("default") {}

    Calculator(const std::string& name) : name(name) {
        history.reserve(10);
    }

    int add(int a, int b) {
        // … method body omitted (4 lines)
    }

    int multiply(int a, int b) {
        // … method body omitted (3 lines)
    }

    std::vector<std::string> getHistory() const {
        return history;
    }

private:
    bool validateInput(int value) {
        // … method body omitted (12 lines)
    }
};

ProcessingResult processUserData(const std::vector<User>& users) {
    // … function body omitted (11 lines)
}

// Template function
template<typename T, typename Processor>
std::vector<T> processArray(const std::vector<T>& items, Processor processor) {
    // … function body omitted (9 lines)
}

// Lambda usage function
std::vector<int> filterPositive(const std::vector<int>& numbers) {
    // … function body omitted (4 lines)
}

int main() {
    // … function body omitted (10 lines)
}
